alloy-core/src/main/java/org/starchartlabs/alloy/core/Preconditions.java

```diff
@@ -205,4 +315,35 @@ public static void checkState(boolean expression, @Nullable Supplier<String> err
         }
     }
 
+    /**
+     * Evaluates a value against a provided {@link Predicate} expression
+     *
+     * <p>
+     * Preferred to {@link #checkArgument(Object, Predicate, Object)} when elements needs to be substituted or otherwise
+     * processed for performance reasons
+     *
+     * @param value
+     *            An assignable value to evaluate
+     * @param predicate
+     *            An operation to evaluate against {@code value}
```

Comment:
The `@param<T>` should be moved to be after the other method arguments

Reply:
moved param<T>